src/python_testing/TC_MCORE_FS_1_4.py

```diff
@@ -0,0 +1,485 @@
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main, type_matches
+from mobly import asserts
+
+
+class Subprocess(threading.Thread):
+
+    def __init__(self, args: list = [], stdout_cb=None, tag="", **kw):
+        super().__init__(**kw)
+        self.tag = f"[{tag}] " if tag else ""
+        self.stdout_cb = stdout_cb
+        self.args = args
+
+    def forward_f(self, f_in, f_out):
+        while True:
+            line = f_in.readline()
+            if not line:
+                break
+            f_out.write(f"{self.tag}{line}")
+            f_out.flush()
+            if self.stdout_cb is not None:
+                self.stdout_cb(line)
+
+    def run(self):
+        logging.info("RUN: %s", " ".join(self.args))
+        self.p = subprocess.Popen(self.args, errors="ignore", stdin=subprocess.PIPE,
+                                  stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        # Forward stdout and stderr with a tag attached.
+        forwarding_stdout_thread = threading.Thread(target=self.forward_f, args=[self.p.stdout, sys.stdout])
+        forwarding_stdout_thread.start()
+        forwarding_stderr_thread = threading.Thread(target=self.forward_f, args=[self.p.stderr, sys.stderr])
+        forwarding_stderr_thread.start()
+        # Wait for the process to finish.
+        self.p.wait()
+        forwarding_stdout_thread.join()
+        forwarding_stderr_thread.join()
+
+    def stop(self):
+        self.p.terminate()
+        self.join()
+
+
+class FabricSyncApp:
+
+    def _process_admin_output(self, line):
+        if self.wait_for_text_text is not None and self.wait_for_text_text in line:
+            self.wait_for_text_event.set()
+
+    def wait_for_text(self, timeout=30):
+        if not self.wait_for_text_event.wait(timeout=timeout):
+            raise Exception(f"Timeout waiting for text: {self.wait_for_text_text}")
+        self.wait_for_text_event.clear()
+        self.wait_for_text_text = None
+
+    def __init__(self, fabric_sync_app_path, fabric_admin_app_path, fabric_bridge_app_path,
+                 storage_dir, fabric_name=None, node_id=None, vendor_id=None,
+                 paa_trust_store_path=None, bridge_port=None, bridge_discriminator=None,
+                 bridge_passcode=None):
+
+        self.wait_for_text_event = threading.Event()
+        self.wait_for_text_text = None
+
+        args = [fabric_sync_app_path]
+        args.append(f"--app-admin={fabric_admin_app_path}")
+        args.append(f"--app-bridge={fabric_bridge_app_path}")
+        # Override default ports, so it will be possible to run
+        # our TH_FSA alongside the DUT_FSA during CI testing.
+        args.append("--app-admin-rpc-port=44000")
+        args.append("--app-bridge-rpc-port=44001")
+        # Keep the storage directory in a temporary location.
+        args.append(f"--storage-dir={storage_dir}")
+        # FIXME: Passing custom PAA store breaks something
```

Comment:
Nit, I like to see github issues for this if we have an intention of fixing it. I am okay with you using https://github.com/project-chip/connectedhomeip/issues/35348

Reply:
I've just checked and it seems that my comment is no longer valid. Passing PAA store path seems to work fine now.